Resources/public/mapbender.element.digitizer.js

```diff
@@ -1723,16 +1726,21 @@
                                 });
                                 return;
                             }
-                            _.extend(dataItem, response.dataItem);
+                            _.extend(dataItem, response);
+                            schema.options[dataItem[uniqueKey]] = dataItem[textKey];
                             if(isNew) {
-                                var textKey = item.dataStore.text;
-                                var uniqueKey = item.dataStore.uniqueId;
 
                                 ref.append('<option value="' + dataItem[uniqueKey] + '">' + dataItem[textKey] + '</option>');
+
+
+                            } else {
```

Comment:
Add case of updating an existing entry in the drop-down menu and update the value for this entry instead of adding a new one